src/avro/src/schema.rs

```diff
@@ -317,14 +318,32 @@ pub enum SchemaPiece {
 impl SchemaPiece {
     /// Returns whether the schema node is "underlyingly" an Int (but possibly a logicalType typedef)
     pub fn is_underlying_int(&self) -> bool {
-        matches!(self, SchemaPiece::Int | SchemaPiece::Date)
+        self.underlying_int_value(0).is_some()
     }
     /// Returns whether the schema node is "underlyingly" an Int64 (but possibly a logicalType typedef)
     pub fn is_underlying_long(&self) -> bool {
-        matches!(
-            self,
-            SchemaPiece::Long | SchemaPiece::TimestampMilli | SchemaPiece::TimestampMicro
-        )
+        self.underlying_long_value(0).is_some()
+    }
+    /// Constructs an `avro::Value` if this is of underlying int type.
+    /// Guaranteed to be `Some` when `is_underlying_int` is `true`.
```

Comment:
I feel like this might be more clearly named `try_make_value_from_int` or somesuch. As currently named it sounds more like an accessor method.